Update README with the latest changes (#25)
skodjob/test-metadata-generator: README.md

# Test documentation generator

> It is a maven-plugin that delivers a set of annotation for your test code. 

Reviewer: ```suggestion
Test documentation generator is a maven-plugin that delivers a set of annotation for your test code. 
```

You can annotate your test classes and test methods and provide details about test scenario.

> Plugin itself then parse the data from the annotations and generates `markdown` files for readable documentation and `fmf` format metadata for automated reporting of the test cases to external systems.

Reviewer: ```suggestion
Plugin itself then parse the data from the annotations and generates `Markdown` files for readable documentation and `fmf` format metadata for automated reporting of the test cases to external systems.
```

## Using the annotations

### Test class usage

To generate documentation of the test class, you can use the following annotations:

> * `@SuiteDoc` - is the main annotation, which consists all other annotation and should be used right above the method.

Reviewer: ```suggestion
* `@SuiteDoc` - is the main annotation, which consists of all other annotation and should be used right above the method.
```

        steps = {
                @Step(value = "Create object instance", expected = "Instance of an object is created"),
                @Step(value = "Do a magic trick", expected = "Magic trick is done with success"),
                @Step(value = "Clean up the test case", expected = "Everything is cleared"),
                @Step(value = "Do a magic cleanup check", expected = "Everything magically work")
        },
        useCases = {
                @UseCase(id = "core"),
                @UseCase(id = "core+"),
                @UseCase(id = "core+++")
        },
        tags = {
                @TestTag(value = "default"),
                @TestTag(value = "regression"),
        }
    )
    void testMyCode(ExtensionContext extensionContext) {
        // ...
    }
```

## Generating the documentation

The generation is handled by maven-plugin that is available at [Maven central](https://central.sonatype.com/artifact/io.skodjob/test-docs-generator-maven-plugin/overview).
All released versions could be found there. 
For every commit we also publish `-SNAPSHOT` version to GitHub [packages](https://github.com/skodjob/test-metadata-generator/packages/2061096).

### Usage

> You can just simply add it to your pom file together with `maven-dependency-plugin` for building the dependencies of the test classes:

Reviewer: ```suggestion
To start using the plugin, you will need to add it to your pom file together with `maven-dependency-plugin` for building the dependencies of the test classes:
```

                </configuration>
            </execution>
        </executions>
    </plugin>
    <plugin>
        <groupId>io.skodjob</groupId>
        <artifactId>test-docs-generator-maven-plugin</artifactId>
        <version>${generator.version}</version>
        <executions>
            <execution>
                <phase>post-integration-test</phase>
                <goals>
                    <goal>test-docs-generator</goal>
                </goals>
            </execution>
        </executions>
        <configuration>
            <filePath>./dummy-module/src/test/java/io/</filePath>
            <generatePath>./docs/</generatePath>
            <generateFmf>true</generateFmf>
        </configuration>
    </plugin>
```

`maven-dependency-plugin` is needed for proper loading the classes. 
Without it the plugin will return `NoClassDefFound` exception and will fail.

### Accepted Parameters

Plugin works with the following parameters:
* `filePath` - path to the built classes, from where all the names of the tests are taken.
* `generatePath` - path to the place where the documentation should be generated.

> * `generateFmf` - boolean value whether generator should generate also fmf metadata or just md.

Reviewer: ```suggestion
* `generateFmf` - boolean value whether generator should generate also `fmf` metadata or just `md`.
```

## Use SNAPSHOT version

> TO use `-SNAPSHOT` versions you have to have the plugin built on your local environment or use GitHub packages for a dependency resolution.

Reviewer: ```suggestion
To use `-SNAPSHOT` versions you have to have the plugin built on your local environment or use GitHub packages for a dependency resolution.
```
